tests/Unit/Utilities/Test_TupleSlice.cpp

```diff
@@ -0,0 +1,91 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <string>
+#include <tuple>
+
+#include "Utilities/TupleSlice.hpp"
+
+namespace {
+
+struct ConstructionObserver {
+  ConstructionObserver() = default;
+  ConstructionObserver(const ConstructionObserver& /*rhs*/) noexcept {
+    status = "copy-constructed";
+  }
+  ConstructionObserver& operator=(
+      const ConstructionObserver& /*rhs*/) noexcept {
+    status = "copied";
+    return *this;
+  };
+  ConstructionObserver(ConstructionObserver&& rhs) noexcept {
+    status = "move-constructed";
+    rhs.status = "move-constructed-away";
+  }
+  ConstructionObserver& operator=(ConstructionObserver&& rhs) noexcept {
+    status = "moved";
+    rhs.status = "moved-away";
+    return *this;
+  };
+  ~ConstructionObserver() = default;
+
+  std::string status = "initial";
+};
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Utilities.TupleSlice", "[Utilities][Unit]") {
+  CHECK(tuple_slice<1, 3>(std::tuple<int, float, double, std::string>{
+            1, 2., 3., ""}) == std::tuple<float, double>{2., 3.});
+  CHECK(tuple_head<2>(std::tuple<int, float, double, std::string>{
+            1, 2., 3., ""}) == std::tuple<int, float>{1, 2.});
+  CHECK(tuple_tail<2>(std::tuple<int, float, double, std::string>{
+            1, 2., 3., ""}) == std::tuple<double, std::string>{3., ""});
+  CHECK(tuple_slice<0, 1>(std::tuple<int>{1}) == std::tuple<int>{1});
+  CHECK(tuple_head<1>(std::tuple<int>{1}) == std::tuple<int>{1});
+  CHECK(tuple_tail<1>(std::tuple<int>{1}) == std::tuple<int>{1});
+  CHECK(tuple_slice<0, 0>(std::tuple<int>{1}) == std::tuple<>{});
+  CHECK(tuple_head<0>(std::tuple<int>{1}) == std::tuple<>{});
+  CHECK(tuple_tail<0>(std::tuple<int>{1}) == std::tuple<>{});
+  CHECK(tuple_slice<0, 0>(std::tuple<>{}) == std::tuple<>{});
+  CHECK(tuple_head<0>(std::tuple<>{}) == std::tuple<>{});
+  CHECK(tuple_tail<0>(std::tuple<>{}) == std::tuple<>{});
+  {
+    INFO("Copy tuple of reference types");
+    ConstructionObserver a{};
+    ConstructionObserver& b = a;
+    const std::tuple<ConstructionObserver, ConstructionObserver&> tuple{a, b};
+    // Assert that tuple_slice behaves like std::tuple_cat
+    static_assert(std::is_same_v<decltype(tuple_slice<0, 2>(tuple)),
+                                 decltype(std::tuple_cat(tuple))>);
+    const auto sliced_tuple = tuple_slice<0, 2>(tuple);
+    CHECK(b.status == "initial");
+    CHECK(std::get<0>(sliced_tuple).status == "copy-constructed");
+    CHECK(std::get<1>(sliced_tuple).status == "initial");
+  }
+  {
+    INFO("Move tuple of reference types");
+    ConstructionObserver a{};
+    ConstructionObserver& b = a;
+    ConstructionObserver c{};
+    std::tuple<ConstructionObserver, ConstructionObserver&,
```

Comment:
what about `const &` and `const &&`? Should those work and be tested?

Reply:
added in f90bd0a